samza-core/src/main/java/org/apache/samza/coordinator/CoordinationConstants.java

```diff
@@ -27,4 +27,7 @@ private CoordinationConstants() {}
   public static final String APPLICATION_RUNNER_PATH_SUFFIX = "ApplicationRunnerData";
   public static final String RUNID_LOCK_ID = "runId";
   public static final int LOCK_TIMEOUT_MS = 300000;
+  public static final String CLUSTERBASED_CONTAINER_HEARTBEAT_SERVELET = "containerHeartbeat";
+  public static final String CLUSTERBASED_EXECUTION_ENVIRONMENT_CONTAINER_ID = "executionContainerId";
+  public static final String CLUSTERBASED_COORDINATOR_URL = "samza.autoscaling.server.url";
```

Comment:
I'd also suggest moving this to Yarn specific constants file and remove the `ClusterBased` prefix as these don't have any implications outside the YARN world as they are all very specific to how YARN operates and aren't used in general samza core orchestration.

[1] `samza.autoscaling.server.url` should have gotten cleaned up when we removed the auto-scaling module. We can take this opportunity and rename this configuration to `am.tracking.url` or something else as this is unused.
[2] If we want to keep it the way it is for backward compatibility reasons, we should add a comment on how this is being used. 

I am in favor of [1]. wdyt?

Reply:
im in favor of [1] too.

agree about these constants being yarn specific but the components using these components (ex: ContainerHeartbeatClient) are part of samza-core. hence making this a yarn specific file in samza-yarn module will cause circular dependency issues.

Additionally, the existing CoordinationConstants file is not labelled standalone specific - although it has only standalone constants. Hence, felt this was a good place to put all constants needed for coordination code

Reply:
Let us leave the constants in the same file. I'd still suggest to remove the `ClusterBased` prefix alone as its only yarn specific and not applicable to a general cluster based deployments like K8s.